knowledge_repo/app/index.py

```diff
@@ -15,19 +15,36 @@
 
 
 def is_indexing():
-    return int(IndexMetadata.get('lock', 'index', '0'))
+    last_update = time_since_index()
+    return bool(int(IndexMetadata.get('lock', 'index', '0'))) and last_update is not None and (last_update < 10 * 60)
```

Comment:
Good call. I want to actually make this configurable (which is perhaps why I was lazy about documentation here)... but if I don't get to do it here, I'll document it as you proposed :).